source-alpaca/main.go

```diff
@@ -0,0 +1,135 @@
+package main
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	boilerplate "github.com/estuary/connectors/source-boilerplate"
+)
+
+const (
+	defaultMaxBackfillInterval = 15 * time.Minute
+	defaultMinBackfillInterval = 1 * time.Minute
+)
+
+type resource struct {
+	Name    string `json:"name" jsonschema:"title=Name,description=Unique name for this binding. Cannot be changed once set."`
```

Comment:
Something about the need for an explicit name here feels like maybe the domain modeling is off. Some random thoughts:

My naive perspective is that a given feed+symbol pairing is its own distinct stream of data which can be captured into a collection, and combining multiple symbols into a single resource binding is the cause of a lot of the weirdness around endpoint/resource configs here. And I'm reasonably sure one can have multiple capture bindings going into the same collection, so no issue there.

Discovery would be troublesome as currently implemented in the UI. But in a model where each stock symbol is a distinct resource there's actually a pretty nice coherent implementation of a discovery UI where we just return a fixed list of some tens of thousands of possible resources and give the user a search box. I think something like that is on the eventual roadmap for other data sources with a ton of different streams?

Reply:
Functionally I think this is similar to how something like the `source-http-file` capture works, where each binding has a "stream", but the stream is just the last part of the provided URL to uniquely identify it.

The thought about having each symbol being a distinct stream is interesting. In order for data collection to be efficient, we'll need to make requests for grouped symbols. I'm not sure how that would work if each binding was responsible for only a single symbol. I'm thinking in the near future we will end up capturing entire exchanges' worth of symbols at a time, and it would definitely make sense to map an exchange to a stream. It will take some work to handle huge amounts of symbols at a time though, so I'm leaving that as a future enhancement for now.